cachi2/core/package_managers/gomod.py

```diff
@@ -140,6 +141,20 @@ def fetch_gomod_source(request: Request) -> RequestOutput:
     )
 
 
+def _dep_to_sbom_component(dep: dict[str, Any]) -> Component:
+    if dep["version"] is not None:
+        return Component(
+            name=dep["name"],
+            version=dep["version"],
+            purl=f"pkg:golang/{dep['name']}@{dep['version']}"
+        )
+
+    return Component(
+        name=dep["name"],
+        purl=f"pkg:golang/{dep['name']}"
+    )
```

Comment:
I just noticed my comment on STONEBLD-930:

> The official CycloneDX gomod implementation uses ?type=module and ?type=package qualifiers
> 
> https://github.com/CycloneDX/cyclonedx-gomod/search?q=%22%3Ftype%3Dmodule%22
> 
> https://github.com/CycloneDX/cyclonedx-gomod/search?q=%22%3Ftype%3Dpackage%22
> 
> Let's use those 🎉

Thoughts?

Reply:
Me gusta!

I saw that yesterday, but for some reason, the links were not working when I clicked them (not they work fine :|)

Reply:
I think we should't report a module twice, as a module and a package though. Since the purls would be different because of the `?type` part, the current dedup would not work.

So `type=package` should be reserved for package-only dependencies. Wdyt?

Reply:
I think we should report both the module and the package. There are also module-only dependencies (none of their packages are reported by `go list -deps ./...`) and that could be valuable information